src/platform/xbox/platform.c

```diff
@@ -408,10 +418,35 @@ void platform_flush_cache()
 // YYYY-MM-DD HH:MM:SS
 void platform_get_iso8601_time(char time_str[20])
 {
-    SYSTEMTIME st;
-    GetLocalTime(&st);
+    static SYSTEMTIME st = {.wHour = 255};
+    static DWORD tick_cnt = 0;
+
+    DWORD change = KeTickCount - tick_cnt;
+    if (change > 1000)
+    {
+        st.wSecond += change / 1000;
+        tick_cnt = KeTickCount - (change % 1000);
+        while (st.wSecond > 59)
+        {
+            st.wMinute++;
+            st.wSecond -= 60;
+        }
+        while (st.wMinute > 59)
+        {
+            st.wHour++;
+            st.wMinute -= 60;
+        }
+    }
+    if (st.wHour > 23)

```

Comment:
At midnight does it actually show `24` for a second and then go `0`? Otherwise this will never hit beyond the first run. Perhaps 23:59:59?

Reply:
Oh yea your right this won't work properly. I'll change it

Reply:
Actually I think it's ok. It can be 24 briefly within the function